docs/en/02_Developer_Guides/15_Customising_the_Admin_Interface/How_Tos/Extend_CMS_Interface.md

```diff
@@ -204,26 +204,443 @@ To make the actions more user-friendly you can also use alternating buttons as
+```js
+import Injector from 'lib/Injector';
+import TextLengthChecker from './components/TextLengthChecker';
+
+Injector.transform(
+  'my-other-transformation', 
+  (update) => update('TextField', TextLengthChecker),
+  { before: 'my-transformation' }
+);
+```
+
+Now, both components, coming from different modules, play together nicely, in the correct
+order.
+
+![](../../../_images/react-di-3.png)
+
+`before` and `after` also accept arrays of constraints.
+
+```js
+Injector.transform(
+  'my-transformation', 
+  (update) => update('TextField', TextLengthChecker),
+  { before: ['my-transformation', 'some-other-transformation'] }
+);
+```
+
+#### Using the * flag
+
+If you really want to be sure your customisation gets loaded first or last, you can use 
+`*` as your `before` or `after` reference. 
+
+```js
+Injector.transform(
+  'my-transformation', 
+  (update) => update('TextField', FinalTransform),
+  { after: '*' }
+);
+```
+**Note**: This flag can only be used once per transformation.
+The following are not allowed:
+* `{ before: ['*', 'something-else'] }`
+* `{ after: '*', before: 'something-else' }`
+
+### Helpful tip: Name your higher order components
+
+Now that we have multiple enhancements happening to the same component, it will be really
+useful for debugging purposes to reveal the names of each enhancement on the `displayName` of
+ the component. This will really help you when viewing the rendered component tree in 
+ [React Dev Tools](https://chrome.google.com/webstore/detail/react-developer-tools/fmkadmapgofadopljbjfkapdkoienihi?hl=en).
+ 
+ For this, you can use the third parameter of the `update()` function. It takes an arbitrary
+ name for the enhancement you're applying.
+ 
+ __module-a/js/main.js__
+ ```js
+ (update) => update('TextField', CharacterCounter, 'CharacterCounter')
+ ```
+ __module-b/js/main.js__
+ ```js
+ (update) => update('TextField', TextLengthChecker, 'TextLengthChecker')
+ ```
+### Dealing with events
+Let's make a new customisation that customises the behaviour of a button. We'll have
+all form actions throw a `window.confirm()` message before executing their action. Further,
+we'll apply some new style to the button if it is in a loading state.
+
+__my-module/js/components/ConfirmingFormButton.js__
+```js
+import React from 'react';
+
+export default (FormAction) => (props) => {
+  const newProps = {
+    ...props,
+    data: {
+      ...props.data,
+      buttonStyle: props.loading ? 'danger' : props.data.buttonStyle
+    },
+    handleClick(e) {
+      if(window.confirm('Did you really mean to click this?')) {
+        props.handleClick(e);
+      }
+    }
+  }
+
+  return <FormAction {...newProps} />
+}
+```
+
+__my-module/js/main.js__
+```js
+import ConfirmingFormButton from './components/ConfirmingFormButton';
+
+Injector.transform('my-transformation', (update) => {
+  update('FormAction', ConfirmingFormButton, 'ConfirmingFormButton');
+});
+```
+### Registering new React components
```

Comment:
Maybe this could be put at the end of the Gallery example at the top? I kind of read that and wondered, "okay, so, now what? I've created a component but I can't use it anywhere."